### PR TITLE
Unifies wield-shotgun and wield-rifle into wield-firearm.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
Title says it all. 

This needed some changes to offhand (which is now a subtype of weapon, not /weapon/gun), but otherwise it seems to work well in testing.

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

**Comment by sheeplets** [comment]
Does this apply to kinetic accelerators as well?

**Comment by NonQueueingMatt** [comment]
Now it does, @sheeplets

File: code/game/objects/items/weapons/vaurca_items.dm [hunk 1]
```diff
@@ -361,22 +361,11 @@
...
-	if(src in usr)
-		toggle_wield(usr)
-
-/obj/item/weapon/gun/launcher/crossbow/vaurca/verb/wield_rifle()
-	set name = "Wield rifle"
-	set category = "Object"
-	set src in usr
-
-	toggle_wield(usr)
+/obj/item/weapon/gun/launcher/crossbow/vaurca/wield()
```
Comment (skull132) [inline code/game/objects/items/weapons/vaurca_items.dm]: also useless

**Comment by NonQueueingMatt** [comment]
!wip

**Comment by NonQueueingMatt** [comment]
!review

**Comment by NonQueueingMatt** [comment]
@skull132 this should be done now

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -49,6 +49,7 @@
 	origin_tech = list(TECH_COMBAT = 1)
 	attack_verb = list("struck", "hit", "bashed")
 	zoomdevicename = "scope"
+	action_button_name = ""
```
Comment (skull132) [inline code/modules/projectiles/gun.dm]: Is this necessary? The default value should do, IMO, ergo, remove the needless override.

File: code/modules/custom_ka/core.dm [hunk 3]
```diff
@@ -28,8 +28,6 @@
 	dispersion = list(0)
 	reliability = 100
```
Comment (skull132) [inline code/modules/custom_ka/core.dm]: Missing `wieldable = TRUE` here?

File: code/modules/projectiles/guns/energy/laser.dm [hunk 4]
```diff
@@ -134,16 +134,6 @@ obj/item/weapon/gun/energy/retro
 	fire_delay_wielded = 35
 	accuracy_wielded = 0
 
-	//action button for wielding
-	action_button_name = "Wield rifle"
```
Comment (skull132) [inline code/modules/projectiles/guns/energy/laser.dm]: Missing `is_wieldable = TRUE` for this rifle?

File: code/modules/projectiles/guns/energy/modular.dm [hunk 5]
```diff
@@ -79,12 +79,10 @@
 			gun_type = CHASSIS_MEDIUM
 			slot_flags = SLOT_BELT | SLOT_BACK
 			item_state = "energystun"
-			action_button_name = "Wield rifle"
```
Comment (skull132) [inline code/modules/projectiles/guns/energy/modular.dm]: Missing `is_wieldable = TRUE` here and below? Though you will also need to make logic to actually add the button in (and remove it in `reset_vars`). You might want to subproc the initial switch as well into `updated_is_wieldable()` or something.

File: code/modules/projectiles/guns/projectile/rifle.dm [hunk 6]
```diff
@@ -19,8 +19,6 @@
 	can_sawoff = TRUE
 	sawnoff_workmsg = "shorten the barrel and stock"
```
Comment (skull132) [inline code/modules/projectiles/guns/projectile/rifle.dm]: Missing `is_wieldable = TRUE` here?

Comment (NonQueueingMatt) [reply code/modules/projectiles/guns/projectile/rifle.dm]: it inherits that from shotgun/pump

**Comment by NonQueueingMatt** [comment]
!wip

**Comment by NonQueueingMatt** [comment]
!review

File: code/modules/projectiles/gun.dm [hunk 7]
```diff
@@ -173,6 +176,29 @@
...
+/obj/item/weapon/gun/ui_action_click()
+	if(src in usr)
+		wield_gun()
+
+/obj/item/weapon/gun/proc/update_wield_verb()
+	if(is_wieldable) //If the gun is marked as wieldable, make the action button appear and add the verb.
+		action_button_name = "Wield Firearm"
+		verbs += /obj/item/weapon/gun/verb/wield_gun
+	else
+		verbs -= /obj/item/weapon/gun/verb/wield_gun
```
Comment (skull132) [inline code/modules/projectiles/gun.dm]:
```suggestion
		action_button_name = ""
		verbs -= /obj/item/weapon/gun/verb/wield_gun
```

Gotta assume this gets called in a fashion like TRUE -> FALSE, at which case, we gotta null this var as well.

**Pull request merged**